dapps-profiling/src/profilings/tokens/index.ts

```diff
@@ -1,167 +1,165 @@
 import { Network } from "stellar-plus/lib/stellar-plus/types";
+import { createBaseAccounts, setupAssets } from "./setup";
+
 import {
-  addTrustlinesToUsers,
-  createBaseAccounts,
   mintSorobanTokensToUsers,
-  setupAssets,
-} from "./setup";
+  setupDemuUsers,
+} from "../../utils/simulation/functions";
+import { DemoUser } from "../../utils/simulation-types";
+import {
+  profileBurn,
+  profileMinting,
+  profilePayments,
+} from "./profiling-simulations";
 import { exportArrayToCSV } from "../../utils/export-to-csv";
+import { Console } from "console";
 
 export type tokensProfilingConfigType = {
   nUsers: number;
   nTransactions: number;
   network: Network;
+  transactions?: tokenTransactions[];
 };
 
+export enum tokenTransactions {
+  transfer = "transfer",
+  // transferFrom = "transferFrom",
+  // approve = "approve",
+  mint = "mint",
+  burn = "burn",
+}
+
+/**
+ * @args {tokensProfilingConfigType} args
+ * @param {number} args.nUsers - Number of users to be created
+ * @param {number} args.nTransactions - Number of transactions to be executed in each testing phase
+ * @param {Network} args.network - Network to be used
+ * @param {tokenTransactions[]} args.transactions - Transactions to be executed. If not provided, all transactions will be executed
+ * If the mint transaction is not provided, the tests are skipped for this transactin but still, the mint transaction will be executed for each user during setup.
+ *
+ * @description This function will execute the following steps:
+ * 1. Create base accounts - opex and issuer accounts. Opex will be used to pay for the transactions, issuer will be used to issue and manage the tokens.
+ * 2. Setup assets - Soroban token and SAC token. Soroban token is a pure soroban token, SAC token is a classic token wrapped in a soroban contract.
+ * 3. Setup users - Create users, add trustlines to the tokens and mint tokens to the users.
+ * 4. Execute transactions - Execute transactions with the tokens and users created in the previous steps. Everything is captured by the profiler for later analysis.
+ * 5. Export results - Export the results to a CSV file.
+ **/
 export const tokensProfiling = async (args: tokensProfilingConfigType) => {
```

Comment:
Hey Fifo, what do you think about destructuring the function arguments, it makes it easier to read and removes all the `args.` from your code:
```suggestion
export const tokensProfiling = async ({ nTransactions, nUsers, network, transactions }: tokensProfilingConfigType) => {
```

Reply:
Adjusted